src/qutip_cupy/dense_functions.py

```diff
@@ -29,3 +29,12 @@ def trace_cupydense(cp_arr):
     # @TODO: whnen qutip allows it we should remove this call to item()
     # as it takes a time penalty commmunicating data from GPU to CPU.
     return cp.trace(cp_arr._cp).item()
+
+
+def pow_cupydense(cp_arr, n):
+    if cp_arr.shape[0] != cp_arr.shape[1]:
+        raise TypeError("matrix power only works with square matrices")
```

Comment:
```suggestion
        raise ValueError("matrix power only works with square matrices")
```
The type is correct, but the shape is not.